### PR TITLE
Apply PHP 8.0 Syntax and constructor promotion

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 8.0, php 8.0 syntax can constuctor promotion can be applied.

### CODE REVIEW

**Comment by samsonasik** [comment]
All green 🎉

File: src/ConfigDiscoveryTrait.php [hunk 1]
```diff
@@ -118,7 +116,7 @@ private function getApplicationConfig()
             return $this->applicationConfig;
         }
 
-        $configFile = isset($this->projectDir)
```
Comment (Ocramius) [inline src/ConfigDiscoveryTrait.php]: This change seems wrong, considering the scope of the trait

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/ConfigDiscoveryTrait.php (hunk 1 above)
Comment (samsonasik) [reply src/ConfigDiscoveryTrait.php]: Ok, rolled back to isset with update psalm config https://github.com/laminas/laminas-development-mode/pull/23/commits/b592b61bed6e5ac4e8faf85ce6dde1b8856b9cce

**Comment by samsonasik** [comment]
All green 🎉

**Review by Ocramius — APPROVED** [review]
Thanks @samsonasik!

**Pull request merged**